Client/Assets/SpotlightHelper.js

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+  * License, v. 2.0. If a copy of the MPL was not distributed with this
+  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function() {
+  var selectors = {
+    title: "head title",
+    description: "head meta[name='description'], body p",
+  };
+
+  function collectText ($document, selector) {
+    var $el = $document.querySelector(selector);
+    if ($el) {
+      return $el.getAttribute("content") || $el.innerText
+    }
+  }
+
+  function assemblePayload ($document, selectors) {
+    var payload = {};
+    for (var key in selectors) {
+      payload[key] = collectText($document, selectors[key]) || "";
+    }
+    return payload;
+  }
+
+  window.addEventListener("load", function() {
+     var payload = assemblePayload(document, selectors);
+     webkit.messageHandlers.spotlightMessageHandler.postMessage(payload);
+  });
+
```

Comment:
I think this is a great first pass at extracting interesting info from a page to display in the search index. I came across some documentation for web developers [1] that enable them to enrich their pages with discoverable content that can be used in spotlight. I wonder if it makes sense in the next iteration to tap into this data to add richer spotlight integration.

[1] https://developer.apple.com/library/ios/documentation/General/Conceptual/AppSearch/WebContent.html#//apple_ref/doc/uid/TP40016308-CH8